libs/motor_control/esc/esc.h

```diff
@@ -0,0 +1,172 @@
+    ((DC_MIN) + (((DC_MAX) - (DC_MIN)) * ((OFFSET) / 100.0f)))
+#define ESC_APPLY_MAX_OFFSET(OFFSET, DC_MIN, DC_MAX)                           \
+    ((DC_MAX) - (((DC_MAX) - (DC_MIN)) * ((OFFSET) / 100.0f)))
+
+// Defined in Kconfig
+#if CONFIG_ESC_MANUAL_MIN_THRUST_OFFSET
+
+#define ESC_DC_MIN_PWM_US                                                      \
+    ESC_APPLY_MIN_OFFSET(CONFIG_ESC_DC_MIN_PWM_US_OFFSET, 1000.0f, 2000.0f)
+#define ESC_DC_MIN_ONESHOT_125_US                                              \
+    ESC_APPLY_MIN_OFFSET(CONFIG_ESC_DC_MIN_PWM_US_OFFSET, 125.0f, 250.0f)
+#define ESC_DC_MIN_ONESHOT_42_US                                               \
+    ESC_APPLY_MIN_OFFSET(CONFIG_ESC_DC_MIN_PWM_US_OFFSET, 42.0f, 84.0f)
+#define ESC_DC_MIN_MULTISHOT_US                                                \
+    ESC_APPLY_MIN_OFFSET(CONFIG_ESC_DC_MIN_PWM_US_OFFSET, 5.0f, 25.0f)
+
+#else
+
+#define ESC_DC_MIN_PWM_US 1000
+#define ESC_DC_MIN_ONESHOT_125_US 125
+#define ESC_DC_MIN_ONESHOT_42_US 42
+#define ESC_DC_MIN_MULTISHOT_US 5
+
+#endif
+
+// Defined in Kconfig
+#if CONFIG_ESC_MANUAL_MAX_THRUST_OFFSET
+
+#define ESC_DC_MAX_PWM_US                                                      \
+    ESC_APPLY_MAX_OFFSET(CONFIG_ESC_DC_MAX_PWM_US_OFFSET, 1000.0f, 2000.0f)
+#define ESC_DC_MAX_ONESHOT_125_US                                              \
+    ESC_APPLY_MAX_OFFSET(CONFIG_ESC_DC_MAX_PWM_US_OFFSET, 125.0f, 250.0f)
+#define ESC_DC_MAX_ONESHOT_42_US                                               \
+    ESC_APPLY_MAX_OFFSET(CONFIG_ESC_DC_MAX_PWM_US_OFFSET, 42.0f, 84.0f)
+#define ESC_DC_MAX_MULTISHOT_US                                                \
+    ESC_APPLY_MAX_OFFSET(CONFIG_ESC_DC_MAX_PWM_US_OFFSET, 5.0f, 25.0f)
+
+#else
+#define ESC_DC_MAX_PWM_US 2000
+#define ESC_DC_MAX_ONESHOT_125_US 250
+#define ESC_DC_MAX_ONESHOT_42_US 84
+#define ESC_DC_MAX_MULTISHOT_US 25
+
+#endif
+
+//  ESC Protocol Period in us
+#define ESC_T_PWM_US 4000
+#define ESC_T_ONESHOT_125 500
+#define ESC_T_ONESHOT_42 168
+#define ESC_T_MULTISHOT 50
+
+// ESC Arming duty cycle [us]
+#define ESC_ARM_DC_PWM_US 1000
+#define ESC_ARM_DC_ONESHOT_125_US 125
+#define ESC_ARM_DC_ONESHOT_42_US 42
+#define ESC_ARM_DC_MULTISHOT_US 5
+#define ESC_ARM_DURATION 3000 // NOTE: Arm duration is around 2000-5000 ms
+
+#define ESC_US_TO_NS(x) ((x) * 1000.0f)
+
+typedef enum {
+    ESC_SUCCESS = 0,
+    ESC_DEVICE_PWM_NOT_READY,
+    ESC_INVALID_PROTOCOL,
+    ESC_INVALID_SPEED_SET,
+    ESC_TIMER_OVERFLOW_ERROR,
+    ESC_NOT_INITIALIZED,
+    ESC_INVALID_STOP,
+    ESC_INVALID_ARMING,
+} ESC_Error_Type;
+
+typedef enum {
+    ESC_PWM = 0,
+    ESC_ONESHOT_125,
+    ESC_ONESHOT_42,
+    ESC_MULTISHOT
+} ESC_Protocol_Type;
+
+typedef struct {
+    uint8_t instance_count;
+    const struct device *pwm_dev;
+    uint32_t pwm_channel;
+    uint32_t period;
+    float dc_min;
+    float dc_max;
+    ESC_Protocol_Type protocol;
+    bool flag;
```

Comment:
Flag is a very unclear variable name, rename this to indicate what it actually represents.